client/server/EngineModel.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const vscode_languageserver_1 = require("vscode-languageserver");
+var fs = require('fs');
+var xml2js = require('xml2js');
+var parser = new xml2js.Parser();
+class EngineModel {
+    constructor(path, completionItems) {
+        this.path = path;
+        this.completionItems = completionItems;
+        this.loadPredefinedObjects();
+    }
+    loadPredefinedObjects() {
+        // load predefined objects
+        if (fs.existsSync(this.path)) {
+            this.parsePredefinedObjects(this.path, this.completionItems);
+        }
+        else {
+            console.warn(`File “${this.path}” doesn't exist.`);
+        }
+    }
+    reload(path, completionItems) {
+        this.path = path;
+        this.completionItems = completionItems;
+        this.loadPredefinedObjects();
+    }
+    parsePredefinedObjects(path, completionItems) {
+        console.log(`Parsing definition file: ${path}`);
+        fs.readFile(path, function (err, data) {
+            parser.parseString(data, function (err, result) {
+                if (err != null || result == null) {
+                    console.error("Something went wrong during YE model import:\n" + err);
```

Comment:
Since the condition is `err != null || result == null`, we might end up logging a `null` error, right?